src/exercises-final/13.js

```diff
@@ -52,14 +52,22 @@ class Rendux extends React.Component {
 }
 
 function withRendux(Component) {
-  const Wrapper = React.forwardRef((props, ref) => (
-    <Rendux.Consumer>
-      {rendux => <Component {...props} rendux={rendux} ref={ref} />}
-    </Rendux.Consumer>
+  class Wrapper extends React.Component {
```

Comment:
Are you sure these changes are necessary?

Reply:
I think so, because without them you don't see the `displayName` in `react-devtools`. I followed the pattern in the [forwarding refs in higher-order components](https://reactjs.org/docs/forwarding-refs.html#forwarding-refs-in-higher-order-components) of the docs.

Reply:
Now the HOC is wrapped in `ForwardedRef`:

![image](https://user-images.githubusercontent.com/9815009/39097693-b864f724-462d-11e8-96a5-ebb40cda5a8f.png)

Reply:
Right, but are the changes to the HOC implemention necessary for that to work?

Reply:
Here's how it looks with only the change to `StatePrinter`:

![image](https://user-images.githubusercontent.com/9815009/39097738-8412ba3c-462e-11e8-8f79-0e5685aaaacc.png)

The HOC `displayName` `withRendux(StatePrinter)` doesn't show. 😢

Reply:
Oh, nevermind, I understand.